src/opnsense/mvc/app/views/OPNsense/Firewall/filter_rule.volt

```diff
@@ -526,13 +545,72 @@
             $categoryFilter.val(currentSelection).selectpicker('refresh');
         });
 
+        // Track if user has actually changed the interface dropdown, or it was the controller
+        let interfaceInitialized = false;
+
         // Populate interface selectpicker
-        $('#interface_select').fetch_options('/api/firewall/filter/get_interface_list');
+        $('#interface_select').fetch_options(
+            '/api/firewall/filter/get_interface_list',
+            {},
+            function (data) {
+                for (const groupKey in data) {
+                    const group = data[groupKey];
+                    group.items = group.items.map(item => {
+                        const count = item.count ?? 0;
+                        const label = (item.label || '');
+                        const subtext = group.label;
+
+                        const bgClassMap = {
+                            floating: 'bg-primary',
+                            group: 'bg-warning',
+                            interface: 'bg-info'
+                        };
+                        const badgeClass = bgClassMap[item.type] || 'bg-info';
+
+                        return {
+                            value: item.value,
+                            label: label,
+                            'data-content': `
+                                <span>
+                                    ${count > 0 ? `<span class="badge badge-sm ${badgeClass}">${count}</span>` : ''}
+                                    ${label}
+                                    <small class="text-muted ms-2"><em>${subtext}</em></small>
+                                </span>
+                            `.trim()
+                        };
+                    });
+                }
+                // Apply url hash if present, defer after layout and style recalculations
+                requestAnimationFrame(() => {
```

Comment:
one more thing: it seems to me that this logic should execute after the `selectpicker` refresh in `opnsense_ui.js` and thus warrants a second callback.